README.md

```diff
@@ -28,3 +28,153 @@ Tentative roadmap:
 
 - VPN API
   - [ ] VPN API for desktop (Linux, Windows, macOS)
+
+# Cross-platform Development
+
+## Building
+
+In Go, you can compile for other target operating system and architecture by specifying the `GOOS` and `GOARCH` environment variables.
+
+<details>
+  <summary>Examples</summary>
+
+MacOS example:
+```
+% GOOS=darwin go build -C x -o ./bin/ ./outline-connectivity 
+% file ./x/bin/outline-connectivity 
+./x/bin/outline-connectivity: Mach-O 64-bit executable x86_64
+```
+
+Linux example:
+```
+% GOOS=linux go build -C x -o ./bin/ ./outline-connectivity 
+% file ./x/bin/outline-connectivity                      
+./x/bin/outline-connectivity: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=n0WfUGLum4Y6OpYxZYuz/lbtEdv_kvyUCd3V_qOqb/CC_6GAQqdy_ebeYTdn99/Tk_G3WpBWi8vxqmIlIuU, with debug_info, not stripped
+```
+
+Windows example:
+```
+% GOOS=windows go build -C x -o ./bin/ ./outline-connectivity 
+% file ./x/bin/outline-connectivity.exe 
+./x/bin/outline-connectivity.exe: PE32+ executable (console) x86-64 (stripped to external PDB), for MS Windows
+```
+</details>
+
+## Running Linux binaries
+
+To run Linux binaries you can use a Linux container via [Podman](https://podman.io/).
+
+### Set up podman
+<details>
+  <summary>Instructions</summary>
+
+[Install Podman](https://podman.io/docs/installation)
+On macOS (once):
+```sh
+brew install podman
+```
+
+Create the podman service VM (once):
+```sh
+podman machine init
+```
+
+Start the VM (after every time it is stopped):
+```sh
+podman machine start
+``` 
+
+You can see it running with `podman machine list`:
+```
+% podman machine list
+NAME                     VM TYPE     CREATED        LAST UP            CPUS        MEMORY      DISK SIZE
+podman-machine-default*  qemu        3 minutes ago  Currently running  1           2.147GB     107.4GB
+```
+
+When you are done with development, you can stop the machine:
+```sh
+podman machine stop
+```
+</details>
+
+### Run
+
+The easiest way is to run using `go run` directly with the `--exec` flag and our convenience tool `run_on_podman.sh`:
+```sh
+GOOS=linux go run -C x -exec "$(pwd)/run_on_podman.sh" ./outline-connectivity
+```
+
+It also works for tests:
+```sh
+GOOS=linux go test -exec "$(pwd)/run_on_podman.sh"  ./...
```

Comment:
👍 This is very useful! Then we might also be able to configure VSCode to run the tests on different platforms.